### DISCUSSION TITLE
tls problem

### DISCUSSION THREAD

**Discussion by nzubarev122** [rabbitmq/amqp091-go#127] [Q&A]
Hello.

https://go.dev/play/p/cuOKquzMu7K
2022/10/11 19:50:50 Failed to connect to RabbitMQ: x509: certificate signed by unknown authority
exit status 1

Please help.
tell me how to connect a certificate in this code so that you can contact the rabbit server with an example?

**Comment by lukebakken** [comment]
A good reference is this project's TLS testing code:
https://github.com/rabbitmq/amqp091-go/blob/main/tls_test.go

Please also see this troubleshooting guide:
https://www.rabbitmq.com/troubleshooting-ssl.html

**Reply by lukebakken** [reply]
@nzubarev122 please see this **complete** example - https://github.com/lukebakken/amqp091-go-127

The Go `consumer` application connects to RabbitMQ via TLS.
